### PR TITLE
use replicaset if configured

### PR DESCRIPTION
**Pull request by Icehunter**
I use a replicaset. Here's a patch to use as well.

**config.json:**
```
{
    "url": "http://localhost:4567",
    "secret": "fae93eb7-f17b-429d-a614-4783d8466517",
    "database": "mongo",
    "mongo": {
        "host": "127.0.0.1,127.0.0.1,127.0.0.1",
        "port": "27017,27018,27019",
        "username": "",
        "password": "",
        "database": "nodebb"
    }
}
```

### CODE REVIEW

**Comment by julianlam** [comment]
Hi @Icehunter, please sign the [Contributor Assignment Agreement](https://docs.google.com/forms/d/1KrniXs3pwK8dMeAZnnQz2D07ne7LjJTYIYwJrpDqgnE/viewform?usp=send_form), so we can take a look at this PR :smile:

**Comment by Icehunter** [comment]
Done :)

File: src/database/mongo.js [hunk 1]
```diff
@@ -73,7 +73,16 @@
...
 		}
 
-		var connString = 'mongodb://' + usernamePassword + nconf.get('mongo:host') + ':' + nconf.get('mongo:port') + '/' + nconf.get('mongo:database');
+		var hosts = nconf.get('mongo:host').split(',');
+		var ports = nconf.get('mongo:port').toString().split(',');
+		var servers = [];
+
+        for (var i = 0; i < hosts.length; i++) {
+            servers.push(hosts[i] + ':' + ports[i]);
+        }
```
Comment (barisusakli) [inline src/database/mongo.js]: Indentation issue here. We use tabs :angel: 
Other than that this looks good.

**Comment by garyx** [comment]
I just want to add that with the 1.3 upgrade you now have to specify the replicaset, we had to change the database line like this for it to work.

     "database": "nodebb?replicaSet=replicaname" 
Instead of 
     "database": "nodebb"

**Pull request merged**